### ISSUE TITLE
calico felix constantly using high cpu

### ISSUE THREAD

**Issue by ryangrahamnc** [projectcalico/felix#2051]
Calico-felix is using 4 cores.

## Expected Behavior
Lower cpu

## Current Behavior
Calico-felix is using 4 cores. If I kill the pod, the new pod does the same thing.
```
top - 10:17:39 up 47 days, 17:56,  3 users,  load average: 11.96, 10.68, 9.93
Tasks: 1245 total,   1 running, 1015 sleeping,   0 stopped,   0 zombie
%Cpu(s): 17.0 us,  6.2 sy,  0.0 ni, 75.5 id,  0.2 wa,  0.0 hi,  1.1 si,  0.0 st
KiB Mem : 65890772 total,  9779048 free, 40981756 used, 15129968 buff/cache
KiB Swap:        0 total,        0 free,        0 used. 24775552 avail Mem 

   PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                               
 16882 root      20   0   76636  48164  21536 S 394.4  0.1   3205:52 calico-node                                                                                                                           
131031 root      20   0 2583596 270120  22436 S 100.7  0.4 845:41.59 ffmpeg                                                                                                                                
100269 ryan      20   0  429604 130000   4664 S  52.5  0.2  13530:59 gvfs-udisks2-vo                                                                                                                       
  1298 root      20   0 14.002g 507888  25576 S  50.5  0.8  19507:44 kubelet                                                                                                                               
  1549 root      20   0 18.233g 181580  12000 S  14.1  0.3   9968:06 dockerd                                                                                                                               
 17101 root      20   0    1332   1136    388 S  11.8  0.0 135:47.31 bird6                                                                                                                                 
     1 root      20   0  514788  17732   4716 S  11.1  0.0   5887:09 systemd                                                                                                                               
 17102 root      20   0    1084    876    388 S   8.2  0.0 125:30.63 bird                                                                                                                                  
  1231 message+  20   0   54868   8012   3116 S   7.5  0.0   3748:45 dbus-daemon                                                                                                                           
  3643 root      20   0  763976 513216  26424 S   6.6  0.8   3934:03 kube-apiserver                                                                                                                        
100413 ryan      20   0  365508   4924   3068 S   5.6  0.0 977:27.43 gsd-housekeepin                                                                                                                       
 20277 root      20   0 21.938g 9.393g   1548 S   4.6 14.9   1075:31 zrepl                                                                                                                                 
 99336 ryan      20   0   86180  15056   4604 S   4.6  0.0 860:49.50 systemd                 
```

```
➜  ~ ps aux | grep calico
USER        PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root      16882  378  0.0  76636 48164 ?        Sl   Jun23 3218:12 calico-node -felix
root      16883  0.0  0.0  57964 32644 ?        Sl   Jun23   0:50 calico-node -confd
root      17101 16.0  0.0   1332  1136 ?        S    Jun23 136:15 bird6 -R -s /var/run/calico/bird6.ctl -d -c /etc/calico/confd/config/bird6.cfg
root      17102 14.8  0.0   1084   876 ?        R    Jun23 125:57 bird -R -s /var/run/calico/bird.ctl -d -c /etc/calico/confd/config/bird.cfg
root      62938  0.0  0.0  21536  1040 pts/3    S+   10:20   0:00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn calico
```

And here are some logs from the container:
https://gist.github.com/ryangrahamnc/31428ef63b744fe77b70fa3e19c97238

The log items that look interesting to me are:
`2019-06-24 14:10:24.460 [WARNING][102] endpoint_mgr.go 519: Failed to configure interface, will retry error=open /proc/sys/net/ipv4/conf/cali03af6702454/rp_filter: no such file or directory`
and
`2019-06-24 14:10:58.660 [INFO][102] int_dataplane.go 765: Finished applying updates to dataplane. msecToApply=2111.2500609999997`

, although I dont know enough about calico to know if theyre actually important.

## Your Environment
- Single node kubernetes (v1.14.1) where the node is the master. 16core/32threads. 100 pods. 
- calico/node:v3.3.6
- etcd: k8s.gcr.io/etcd:3.3.10
- Linux ryan-desktop 4.18.0-18-generic #19~18.04.1-Ubuntu SMP Fri Apr 5 10:22:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
- I also tend to use Service.ExternalName to share services between namespaces. Not sure if thats important

**Comment by ryangrahamnc** [comment]
Also if I do `vnstat --iflist`, I get 1800 items back. Not sure if thats related, or if vnstat just hasnt cleared out the deleted items.

**Comment by caseydavenport** [comment]
> ip link show | grep cali | wc -l is also 1800

This is pretty fishy given there are only 100 pods on that node. Seems like we might be leaking interfaces somehow. 

I'd check to see if there are any CNI plugin logs which indicate a leak of some sort or if there's something trying to manage interfaces on the machine (CNI plugin logs are emitted to stderr and will show up in whatever container runtime you're using).

**Comment by ryangrahamnc** [comment]
I suspect I butchered my install somehow. Checking though everything, I was missing the calico-kube-controllers deployment. Not sure how that could have happened apart from an error happening on install that I didnt notice.
So I followed the guide to upgrade, then did `modprobe -r ipip` (not sure if that matters, cant really tell if that persists past restarts), then restarted the node, and now its back to 100 cali network interfaces, and hasnt increased for a few days. And calico-node is using hardly any cpu anymore.
